src/working_files.cc

```diff
@@ -346,10 +346,15 @@ Position WorkingFile::getCompletionPosition(Position pos, std::string *filter,
   while (i > 0 && isIdentifierBody(buffer_content[i - 1]))
     --i;
 
-  *replace_end_pos = pos;
-  for (int i = start;
-       i < buffer_content.size() && isIdentifierBody(buffer_content[i]); i++)
-    replace_end_pos->character++;
+  if (g_config->completion.overwriteFollowingIdentifier) {
+    *replace_end_pos = pos;
+    for (int i = start;
+         i < buffer_content.size() && isIdentifierBody(buffer_content[i]); i++)
+      replace_end_pos->character++;
+  } else {
+    // do not overwrite the following identifier
+    *replace_end_pos = pos;
```

Comment:
This could be pulled out before the if-statement, then you don't need the else block at all.

Reply:
Done. Thanks for spotting that.